dripline/core/request_sender.py

```diff
@@ -0,0 +1,132 @@
+__all__ = []
+
+import scarab
+
+from _dripline.core import op_t, Core, Receiver, MsgRequest, MsgReply, DriplineError
+
+import uuid
+import logging
+logger = logging.getLogger(__name__)
+
+__all__.append("RequestSender")
+class RequestSender():
+    '''
+    A mixin class that provides convenient methods for dripline interactions in a dripline objects.
+    Intended for use as a dripline client in scripts or interactive sessions.
+    '''
+    def __init__(self, sender: Core, timeout_s: int=10, confirm_retcodes: bool=True):
+        '''
+        Configures an interface with the necessary parameters.
+
+        Parameters
+        ----------
+            sender : Core
+                Provide a Core object which will use this mixin. This object will be the one to actually send the messages as it implements the Core functions
+            timeout_s: int, optional
+                Time to wait for a reply, in seconds -- default is 10 s
+            confirm_retcodes: bool, optional  
+                If True, and if a reply is received with retcode != 0, raises an exception -- default is True          
+        '''
+        self.sender = sender
+        self._confirm_retcode = confirm_retcodes
+        self.timeout_s = timeout_s
+        self._receiver = Receiver()
+
+
+    def _check_lockout_key(self, lockout_key:str | uuid.UUID =None):
+        nilkey = uuid.UUID('00000000-0000-0000-0000-000000000000')
+        if lockout_key is None:
+            return nilkey
+        if type(lockout_key) == uuid.UUID:
+            return lockout_key
+        try:
+            return uuid.UUID(lockout_key)
+        except ValueError:
+            logger.warning("Lockout Key '{}' is not properly uuid formatted. Defaulting to nilkey. ".format(lockout_key))
+            return nilkey
+
+
+    def _send_request(self, msgop, target, specifier=None, payload=None, timeout=None, lockout_key=None):
+        '''
+        internal helper method to standardize sending request messages
+        '''
+        a_specifier = specifier if specifier is not None else ""
+        a_request = MsgRequest.create(payload=scarab.to_param(payload), msg_op=msgop, routing_key=target, specifier=a_specifier)
+        a_request.lockout_key = self._check_lockout_key(lockout_key)
```

Comment:
As discussed on Slack, this should be changed to:
```
if lockout_key is not None:
    a_request.lockout_key = lockout_key
```

And `_check_lockout_key()` can be removed.